main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/oncall-scheduler/cmd"
+	"github.com/giantswarm/oncall-scheduler/pkg/project"
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	err := mainE(context.Background())
+	if err != nil {
+		panic(fmt.Sprintf("%#v\n", err))
+	}
+}
+
+func mainE(ctx context.Context) error {
+	var err error
+
+	var logger micrologger.Logger
+	{
+		c := micrologger.Config{}
+
+		logger, err = micrologger.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var rootCommand *cobra.Command
+	{
+		c := cmd.Config{
+			Logger: logger,
+
+			GitCommit: project.GitSHA(),
+			Source:    project.Source(),
+		}
+
+		rootCommand, err = cmd.New(c)
+	}
+
+	err = rootCommand.Execute()
+	if err != nil {
+		logger.LogCtx(ctx, "level", "error", "message", "failed to execute command", "stack", microerror.Stack(err))
+		os.Exit(1)
```

Comment:
The reason for `mainE` is that we can return an error and do logging and panic and fatal and exit in `main`. Let's just do the usual error handling here.